FirebaseFunctionsSwift/Sources/Codable/Callable+Codable.swift

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import FirebaseFunctions
+import FirebaseSharedSwift
+
+public extension Functions {
+  /**
+   * Creates a reference to the Callable HTTPS trigger with the given name, the type of an `Encodable`
+   * request and the type of a `Decodable` response.
+   *
+   * - Parameter name: The name of the Callable HTTPS trigger
+   * - Parameter requestType: The type of the `Encodable` entity to use for requests to this `Callable`
+   * - Parameter responseType: The type of the `Decodable` entity to use for responses from this `Callable`
+   */
+  func httpsCallable<Request: Encodable,
+    Response: Decodable>(_ name: String,
+                         requestAs requestType: Request.Type = Request.self,
+                         responseAs responseType: Response.Type = Response.self,
+                         encoder: StructureEncoder = StructureEncoder(),
```

Comment:
Just a quick question: With default parameters for `requestAs` and `responseAs` (and neat that you can actually default them in that fashion - I did not know that), do you need the overload on line 52 too? It appears to be doing the same thing as the default parameters...

Reply:
Good catch! Removed.